src/core/ui.js

```diff
@@ -130,6 +134,30 @@ function createWarnButton(butName, arr, title) {
   ariaDecorate(button, ariaMap);
   return button;
 }
+window.addEventListener("load", () => {
+  ui.addCommand(`Close`, ui.closeModal, "Esc", "❌");
+  menuLength = menu.children.length;
+});
+
+menu.addEventListener("keydown", e => {
+  if (
+    e.target == menu.getElementsByTagName("li")[0].firstElementChild &&
```

Comment:
you can probably just cache the first and last children in variables, or just look them up more cleanly via a CSS selector (e.g:, `menu.querySelector("li:first-child")`)

Reply:
if we do this right, we should be able to use the same code for the other popup dialogs. Basically, there should be a generic trap focus function that traps the focus inside a given context until some action is performed.

Reply:
@marcoscaceres I have made the changes, is it right now or changes are needed

Reply:
Appreciate your efforts. This is a really hard problem and definitely not "good first issue" territory. I think we are getting closer to a solution however. It might be worth Googling a bit more, however as this has to be solved problem. We just need to adapt someone else's solution.